File: src/1-100/1. Two Sum/Solution.js
// More Efficient Solution
/**
 * @param { number[] } nums
 * @param { number } target
 * @return { number[] }
 */
const twoSum = function (nums, target) {
	const numsMap = new Map()

	for (let i = 0; i < nums.length; i++) {
		let comp = target - nums[i]

		if (numsMap.has(comp)) return [ numsMap.get(comp), i ]
		numsMap.set(nums[i], i)
	}

	return []
}


// Less Efficient Solution
/**
 * @param { number[] } nums
 * @param { number } target
 * @return { number[] }
 */
const twoSum2 = function (nums, target) {
	for (let i = 0; i < nums.length - 1; i++) {
		for (let j = i + 1; j < nums.length; j++) {
			if (i !== j && nums[i] + nums[j] === target) {
				return [ i, j ]
			}
		}
	}

	return []
}

File: src/1-100/1. Two Sum/Solution.py
from typing import List


# More Efficient Solution
class Solution:
    @staticmethod
    def twoSum(nums: List[int], target: int) -> List[int]:
        numsMap = {}

        for i in range(len(nums)):
            comp = target - nums[i]

            if comp in numsMap:
                return [numsMap[comp], i]

            numsMap[nums[i]] = i

        return []


# Less Efficient Solution
class Solution2:
    @staticmethod
    def twoSum(nums: List[int], target: int) -> List[int]:
        for i in range(len(nums) - 1):
            for j in range(i + 1, len(nums)):
                if i != j and nums[i] + nums[j] == target:
                    return [i, j]

        return []
